### PR TITLE
activesupport

### PR DESCRIPTION
**Pull request by inveterateliterate**

### CODE REVIEW

File: rails_util.gemspec [hunk 1]
```diff
@@ -15,6 +15,6 @@ Gem::Specification.new do |s|
   s.files       = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
 
   s.add_development_dependency 'rspec'
-  s.add_dependency 'activesupport', '~> 5.0.0'
+  s.add_dependency 'activesupport', '~> 5.2'
```
Comment (inveterateliterate) [inline rails_util.gemspec:18]: should I make this '~> 5' ? figure most projects are rails 5 or above?

Comment (bhennes2) [reply rails_util.gemspec:18]: I think we'd want to preserve backwards compatibility as best we can. Seems like `~> 5` is a safer way to go.

Comment (inveterateliterate) [reply rails_util.gemspec]: agreed! made that change

Comment (dpikt) [reply rails_util.gemspec]: Should the version range be omitted in that case? I think `~> 5` would equivalent to any version.

Comment (inveterateliterate) [reply rails_util.gemspec]: I believe there's a version 6 out there which would be for rails 6.

File: lib/rails_util/timezone_helper.rb [hunk 2]
```diff
@@ -26,7 +26,8 @@ def convert_timezone(time, to_timezone, from_timezone, **options)
       # @param [String] to_timezone the `ActiveSupport::TimeZone` `timezone.tzinfo.name` of the desired timezone
       # @param [String] to_timezone the `ActiveSupport::TimeZone` `timezone.tzinfo.name` of the timezone being converted
       # @return ActiveSupport::Duration object as the number of seconds between the given timezones
-      def offset_in_seconds(time, to_timezone, from_timezone)
+      def offset_in_seconds(to_timezone, from_timezone)
+        time = Time.now
```
Comment (dpikt) [inline lib/rails_util/timezone_helper.rb:30]: Does this fix a bug or is it just a housekeeping change?

Comment (inveterateliterate) [reply lib/rails_util/timezone_helper.rb:30]: When working locally, when calling `utc_offset` on my `time` object was giving me the wrong object (not taking into account daylight savings). That was the whole thing I was trying to fix in this PR. for whatever reason calling `.utc_offset` on `Time.now` is grabbing the correct offset with daylight savings. I found a stackoverflow about it and should have linked it.

Comment (dpikt) [reply lib/rails_util/timezone_helper.rb:30]: Ah gotcha. Is that something you could add a spec to cover? Or was it a weird ephemeral thing?

Comment (inveterateliterate) [reply lib/rails_util/timezone_helper.rb:30]: it's an ephemeral thing but there is this spec that was trying to partially catch this: https://github.com/LaunchPadLab/rails_util/blob/883a05dd2766eaf2d6c2e56dcf65ed771733a78f/spec/rails_util/timezone_helper_spec.rb#L31

Comment (inveterateliterate) [reply lib/rails_util/timezone_helper.rb:30]: I think this helps points to the difference: https://rails.lighthouseapp.com/projects/8994/tickets/3037-timecurrentutc_offset-and-timezoneutc_offset-are-different-during-dst

In Empower I added this updated code locally, and in my tests, I was providing timezones for my time objects, which made them TimeWithZone objects. calling `.utc_offset` on these provide different values during DST then simple time objects, and my specs were failing.

Using `Time.now` in this module to only capture the offset to me builds confidence that DST is properly accounted for.

Comment (dpikt) [reply lib/rails_util/timezone_helper.rb:30]: 👍now that this change has been made, can the mock for `utc_offset` be removed?

Comment (inveterateliterate) [reply lib/rails_util/timezone_helper.rb:30]: it's not mocking it - it's just grabing the offset for the timezone in question (to_timezone) and converting it to hours. `offset_in_seconds` is the delta between the timezones you're converting. I renamed the methods to make that more apparent.

File: spec/rails_util/timezone_helper_spec.rb [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 describe RailsUtil::TimezoneHelper do
-  let(:to_timezone) { 'America/Chicago' } # currently 5 hours behind UTC
-  let(:from_timezone) { 'America/New_York' } # currently 4 hours behind UTC
-  let(:utc_timezone) { 'Etc/UTC' }
+  let(:to_timezone) { 'America/Chicago' }
+  let(:from_timezone) { 'America/New_York' }
+  let(:seconds_per_hour) { 3600 }
 
+  let(:chicago_offset) do
+    offset_str = (TZInfo::Timezone.get(to_timezone).current_period.utc_total_offset / seconds_per_hour).to_s
```
Comment (dpikt) [inline spec/rails_util/timezone_helper_spec.rb:9]: Comment about what this is doing? 😇

Comment (inveterateliterate) [reply spec/rails_util/timezone_helper_spec.rb:10]: added!

**Pull request merged**